cdist: init at 7.0.0

KAction opened the pull request:
Commit-Identity: e4ce8a84-0828-4a22-bc1b-fda5e2ff8538

Commit:
cdist: init at 7.0.0

fabaff commented on pkgs/tools/admin/cdist/default.nix:
```diff
@@ -0,0 +1,51 @@
+    domain = "code.ungleich.ch";
+    owner = "ungleich-public";
+    repo = "cdist";
+    rev = "be6e7fcc08d5c4e98f7ce30b157ee087010cd126";
```
At first glance it wooks like that the source is tagged.

```suggestion
    rev = "${version}";
```

fabaff commented on pkgs/tools/admin/cdist/default.nix:
```diff
@@ -0,0 +1,51 @@
+
+  meta = with lib; {
+    description = "Minimalistic configuration management system";
+    homepage = "https://sdi.st";
```
I get a 404 with that URL.

```suggestion
    homepage = "https://www.cdi.st/";
```

fabaff commented on pkgs/tools/admin/cdist/default.nix:
```diff
@@ -0,0 +1,51 @@
+  meta = with lib; {
+    description = "Minimalistic configuration management system";
+    homepage = "https://sdi.st";
+    changelog = "https://code.ungleich.ch/ungleich-public/cdist/src/branch/master/docs/changelog";
```
```suggestion
    changelog = "https://code.ungleich.ch/ungleich-public/cdist/src/tag/$version/docs/changelog";
```

fabaff commented on pkgs/tools/admin/cdist/default.nix:
```diff
@@ -0,0 +1,51 @@
+    # Mostly. There are still couple types that are gpl3-only.
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ kaction ];
+    platforms = lib.platforms.unix;
```
```suggestion
    platforms = platforms.unix;
```

fabaff commented on pkgs/tools/admin/cdist/default.nix:
```diff
@@ -0,0 +1,51 @@
+  #
+  # When "machine_type" explorer figures out that it is running inside
+  # chroot, it assumes that it has enough privileges to escape it.
+  doCheck = false;
```
Looks like that we could run `pythonImportsCheck`.

fabaff commented on pkgs/tools/admin/cdist/default.nix:
```diff
@@ -0,0 +1,51 @@
+{ lib, buildPythonApplication, fetchFromGitea, sphinxHook, sphinx-rtd-theme }:
```
One item per line make the diffs easier to read in the future.

```suggestion
{ lib
, buildPythonApplication
, fetchFromGitea
, sphinxHook
, sphinx-rtd-theme
}:
```

fabaff reviewed (commented):
Upstream says in their documentation that it could also be used as a [library](https://www.cdi.st/manual/latest/cdist-integration.html). I'm not sure how common that use-case is.

KAction commented:
I think I addressed all comments. Integration library is still experimental, according to upstream:

> WARNING: cdist integration helper functions are not yet stable!

So I'd rather not split binary and library until somebody comes and say that he wants it.

KAction commented:
@fabaff Can you take another look, please?

KAction commented:
@happysalada Can you take a look, please?
